src/interface/meta.thrift

```diff
@@ -145,9 +155,10 @@ struct CreateTagReq {
 }
 
 struct AlterTagReq {
-    1: common.GraphSpaceID    space_id,
-    2: string                 tag_name,
-    3: list<AlterSchemaItem>  tag_items,
+    1: common.GraphSpaceID      space_id,
+    2: string                   tag_name,
+    3: list<AlterSchemaOption>  schema_options,
+    4: list<AlterSchemaProp>    schema_props,
```

Comment:
Why not 

struct AlterTagReq {
      1: common.GraphSpaceID    space_id,
      2: string                 tag_name,
      3: list<AlterSchemaItem>  tag_items,
      4: SchemaProp schema_prop,
}

?

Reply:
For example,  `alter tag t add (col1 int, col2 int) ttl_col=col1  ttl_duration=100;`

` add (col1 int, col2 int) `  is a `AlterSchemaOption`,  each `AlterSchemaOption` may contain many columns and the type to be operated.

`ttl_col=col1`and  `ttl_duration=100` are two `AlterSchemaProp`.   each `AlterSchemaProp` may  the schema propery type and value.

Reply:
So I think that `AlterSchemaOption` and `AlterSchemaProp` need to be defined separately

Reply:
I notice all schema props have been defined in SchemaProp,   you could set the props you really want to change through it,  right?

Reply:
Yeah，👍
when ttl_col is not empty, then alter sentence is like :
`alter tag t  ttl_duration=100;`

We can just change a shema property through the alter statement.

Reply:
I mean we don't need the structure "AlterSchemaOption/AlterSchemaProp".   SchemaProp is enough here.

Reply:
First, the `tag_items` you said here is `AlterSchemaOption`

Second，the `SchemaProp` and `Schema` are defined as follow：
```
struct SchemaProp {
   1: i64      ttl_duration,
    2: string   ttl_col,
}

 struct Schema {
    1: list<ColumnDef> columns,	
    2: SchemaProp schema_prop,
}
```
`AlterSchemaProp`  only changes ttl_col and ttl_duration;
 `AlterSchemaOption` can add、chage、 drop multi columns.

So, I think we need the structure "AlterSchemaOption/AlterSchemaProp".

Reply:
`alter tag t add (col1 int, col2 int)`  is only used `AlterSchemaOption`,  not involving `SchemaProp` 
`alter tag t ttl_duration=100;`  is only used `AlterSchemaProp `， involving `SchemaProp`

Reply:
I mean the definition looks like
```
struct AlterTagReq {
1: common.GraphSpaceID space_id,
2: string tag_name,
3: list< AlterSchemaItem> tag_items,
4: SchemaProp schema_prop,
}
```
For "alter tag t add (col1 int, col2 int)",   schema_prop is not set.
For "alter tag t ttl_duration=100",  only schema_prop. ttl_duration is involved.

Reply:
With this definition,  the pr should be much simpler.

Reply:
Your definition is Equivalent to:
```
struct AlterTagReq {
1: common.GraphSpaceID space_id,
2: string tag_name,
3: list<AlterSchemaOption>  schema_options  (list<AlterSchemaItem> tag_items)
4: SchemaProp schema_prop,
}
```
Now our argument is only in item 4.
Consider the following sql：
` alter tag t  ttl_duration=100;`
In the alter statement we only know the value of ttl_duration, we don't know the value of other attributes.
i.e. ttl_col.

So when we use the alter statement to update the value of the schema property, we only update the value of ttl_duration, the other options（i.e. ttl_col) retain the original value.

`This idea is the same as updating the columns in the schema. We don't know all the original values in the schema attribute in the alter statement. We only know the value we want to change.`

Reply:
`SchemaProp` represents all schema attributes (i.e. `ttl_duration` and `ttl_col`)
`AlterSchemaProp` only represents one of the schema attributes (i.e. ttl-`duration` or `ttl_col`)

Reply:
You could only set the attribute changed in SchemaProp.

Reply:
@dangleptr ， Let's discuss offline, I will call you. Thx